src/components/Form/СreateUserAccountForm/CreateUserAccountForm.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useMemo, useState } from 'react';
+import { Box, TextField, Typography } from '@material-ui/core';
+import i18n from 'i18next';
+import { langTokens } from '../../../locales/localizationInit';
+import { UserAccountButton } from '../Buttons';
+import { useStyles } from './styles/CreateUserAccountForm.style';
+import { regex } from '../../../views/Profile/PersonalInfo/constants/regex';
+import { UserEmailType } from '../../../views/Profile/PersonalInfo/types';
+
+type CreateUserAccountFormPropsType = {
+  usersEmails: UserEmailType;
+  onClick: (
+    e: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+    email: string,
+  ) => void;
+  currentUserEmail?: string;
+  isLoading?: boolean;
+};
+type ErrorFieldsType = {
```

Comment:
just a thought: I think we could use composite `StateType = { email: string }` so we could re-use it in `ErrorFieldsType` and `VisitedFieldsType` as mapped-type thing:

```
type ErrorFieldsType = {
    [property in keyof StateType]: string
};

type VisitedFieldsType = {
    [property in keyof StateType]: boolean
};
```

just to avoid hard-coding the same structure again and again. 
Moreover, we can specify type based on initial value:

```
const initialValues = {
  email: ''
};

type ErrorFieldsType = {
  [prop in keyof typeof initialState]: string
};

...
  ... =  useState({ ...initialState });
```

something like this. However, I'm not sure if it worth it here and I don't think this pattern is used anywhere else in the project

Reply:
I think we can change it

Reply:
with current version of code `initialValues` constant is nowhere used. And I guess it will be more confusing to the future reader than having hard-coded types here and there(I mean `email: string`, `email: boolean` in initial version)

let's revert to initial version. Approach I proposed might make sense with many fields, but does not align with single `useState(primitiveStringValue)`